### PR TITLE
Describe procedure to add new alert notification medium

### PR DESCRIPTION
**Pull request by anmolbabu**
tendrl-bug-id: Tendrl/specifications#242
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by anmolbabu** [comment]
@nthomas-redhat @brainfunked @shtripat @GowthamShanmugam Please review

I don't know much specific to snmp so I have written this spec generic to any notification method

File: specs/adding_new_alert_notification_medium.adoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+https://github.com/Tendrl/notifier/blob/master/tendrl/alerting/notification/__init__.py#L28
+----
+
+Following are the functions to be implemented in the new subclass:
+
+* set_destinations: This would let the plugin know to all the destinations
+  specific to the plugin
+* get_alert_destinations: Logic to get refined list of destinations based on
+  current alert notification subscriptions.
+* format_message: If any custom formatting custom to plugin needs to be made
```
Comment (shtripat) [inline specs/adding_new_alert_notification_medium.adoc:38]: Correct the grammar of sentence

File: specs/adding_new_alert_notification_medium.adoc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  in addition to the formatted readable message in alert.
+
+  ----
+  Note: The example for custom formatting the message custom to plugin is as in
+  https://github.com/Tendrl/notifier/blob/master/tendrl/alerting/notification/handlers/mail_handler.py#L86
```
Comment (shtripat) [inline specs/adding_new_alert_notification_medium.adoc:43]: This link could be to notifier project

File: specs/adding_new_alert_notification_medium.adoc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  ----
+
+* dispatch_notification: Logic to send out the alert..
+
+The call to dispatch_notification is made automatically whenever there is a
+new alert detected. But, calls to other functions currently are left to
+implementation of dispatch_notification. The framework can be improved to
+automatically automate the invocation of all these plugin functions.
```
Comment (shtripat) [inline specs/adding_new_alert_notification_medium.adoc:51]: s/improved to automatically automate/improved to automate/

File: specs/adding_new_alert_notification_medium.adoc [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+Finally the plugin code written based on the information above needs to be
+placed in the path tendrl/alerting/notification/handlers(will change to
```
Comment (shtripat) [inline specs/adding_new_alert_notification_medium.adoc:54]: `tendrl/notifier/notification/handlers`

File: specs/adding_new_alert_notification_medium.adoc [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+=== Data model impact:
+
+None.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* An Api to configure the notification plugin(channel) specific end-points
```
Comment (shtripat) [inline specs/adding_new_alert_notification_medium.adoc:76]: Detail this a bit more with payload etc

File: specs/adding_new_alert_notification_medium.adoc (hunk 2 above)
Comment (anmolbabu) [reply specs/adding_new_alert_notification_medium.adoc:43]: Yeah it is to the notifier module but currently refactoring PR by @GowthamShanmugam is under review So I thought of changing once that's in

**Pull request closed without merging**